gms/gossiper.cc

```diff
@@ -2017,6 +2018,31 @@ future<> gossiper::add_local_application_state(std::initializer_list<std::pair<a
     })));
 }
 
+future<> gossiper::update_quarantined_hosts() {
+    bool changed = false;
+    for (const auto& [ep, eps] : _endpoint_state_map) {
+        auto asp = eps.get_application_state_ptr(gms::application_state::QUARANTINED_HOSTS);
+        if (asp) {
+            auto hosts_ids = versioned_value::host_ids_from_string(asp->value);
+            logger.debug("update_quarantined_hosts: {} -> {}", ep, asp->value);
+            for (const auto& host_id : hosts_ids) {
+                if (!host_id) {
+                    on_internal_error(logger, "Null quarantine host");
+                }
+                auto [_, inserted] = _quarantined_hosts.insert(host_id);
+                if (inserted) {
+                    changed = true;
+                    logger.debug("update_quarantined_hosts: {}", host_id);
+                    co_await _sys_ks.local().quarantine_host(host_id);
+                }
+            }
+        }
+    }
+    if (changed) {
+        auto& local_state = get_endpoint_state(get_broadcast_address());
+        local_state.add_application_state(gms::application_state::QUARANTINED_HOSTS, versioned_value::host_ids(_quarantined_hosts));
```

Comment:
good point, will do

Reply:
Hmm, rethinking about that, if we move the accumulation of _quarantined_hosts out to storage_service as you suggested, this might simplify this mechanism a lot, and storage_service will update the gossiper local application state only when it learns about a new quarantined host from another node.

Reply:
_quarantined_hosts moved to storage_service in v3.